### PR TITLE
Reorganize repo to align with a few guiding practices

### PR DESCRIPTION
**Pull request by ohaibbq**
## Description of the change

This updates our repository structure to more closely align with the notion of being a component library, rather than a mock app.

- Styles should reside alongside components
- Stories / types should reside alongside components
- Multiple styled components should be consolidated into a single styles file
- Assets should live outside the components directory

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Configuration change (adjusts configuration to achieve some end related to functionality, development, performance, or security)

### Code review

These boxes should be checked by reviewers prior to merging:

- [x] This pull request has a descriptive title and information useful to a reviewer
- [x] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [x] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

File: src/components/CaseloadHealthCard/CaseloadHealthCard.styles.tsx [hunk 1]
```diff
@@ -0,0 +1,88 @@
+// Recidiviz - a data platform for criminal justice reform
+// Copyright (C) 2020 Recidiviz, Inc.
```
Comment (macfarlandian) [inline src/components/CaseloadHealthCard/CaseloadHealthCard.styles.tsx:2]: 2021 boyo

**Review by macfarlandian — APPROVED** [review]
overall I think this scheme makes a lot of sense! 👍🏻 

one note, in other repos we have put our ambient type declarations in an `@types` directory outside of `src`, which I gather is at least somewhat conventional? not something I have enormously strong feelings about though

**Comment by macfarlandian** [comment]
@ohaibbq FWIW I cannot actually run storybook locally in this repo ... build fails because the dependencies from `@fontsource` are missing. needs a package.json update maybe?

**Comment by ohaibbq** [comment]
Too many changes in this diff! #39 has the updated `package.json`. Excited to get back too a clean changeset for some more concise PRs in the near future.

**Pull request merged**